### PR TITLE
 replace jetbrains NotNull/Nullable annotations with jakarta

### PR DESCRIPTION
**Pull request by todvora**
The jetbrains annotations come most likely from method extractions in IDEA. The type of annotation is configurable:

image

Since this appears quite often in code reviews, we should standardize and unify usage of Nullable/NotNull annotations. This PR replaces all usages with jakarta counterparts. Additionally there are now two more forbiddenapis rules blocking the jetbrains annotations.

/jpd Graylog2/graylog-plugin-enterprise#7039

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -845,6 +845,9 @@
 
                             @defaultMessage Use jakarta.validation.
                             javax.validation.**
+
+                            org.jetbrains.annotations.NotNull @ Use jakarta.validation.constraints.NotNull
```
Comment (thll) [inline pom.xml:849]:
```suggestion
                            org.jetbrains.annotations.NotNull @ Use jakarta.annotation.Nonnull
```

**Review by thll — CHANGES_REQUESTED** [review]
Thanks for the initiative!

`jakarta.validation.constraints.NotNull` is used for bean validation. We shouldn't use it to replace `org.jetbrains.annotations.NotNull`.

We should use `jakarta.annotation.Nonnull` instead.

File: pom.xml (hunk 1 above)
Comment (todvora) [reply pom.xml:849]: Thanks, good point, let me change it.

**Comment by todvora** [comment]
@thll There are still many javax.annotation.Nullable and javax.annotation.Nonnull annotations, which we could replace as well. But rather in a different PR, to avoid mixing too many things together.

**Comment by thll** [comment]
> @thll There are still many javax.annotation.Nullable and javax.annotation.Nonnull annotations, which we could replace as well. But rather in a different PR, to avoid mixing too many things together.

@todvora I agree. Let's focus on the jetbrains annotations here.

**Pull request merged**